### ISSUE TITLE
No info about unused "qualifiedBy"

### ISSUE THREAD

**Issue by maposchmi** [mapstruct/mapstruct#797]
Sample:
```
@Mapper
public interface AB2AX
{
    @Mapping(source = "b", target = "x", qualifiedBy = TakeThis.class)
    AX map(AB ab);
 
    X fittingMapping(B b);
 
    @TakeThis
    ZZZ unusableMapping(B b);
}
 
class AB {A a; B b; // getter, setter}
class AX {A a; X x; // getter, setter}
// class A, B, X, ZZZ ...
```

The generated Mapping uses "fittingMapping".
The "qualifiedBy" is silently ignored.
We would expect at least a warning, or even a compile error.

**Comment by maposchmi** [comment]
Similar to https://github.com/mapstruct/mapstruct/issues/767

**Comment by agudian** [comment]
@maposchmi thanks for the report! I agree that it's very much the essence of what needs to be done in #767 - so let's close this as a duplicate and see that we fix #767.
